projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-checkout-flow.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Jetpack_WooCommerce_Analytics_Checkout_Flow
+ *
+ * @package automattic/jetpack
+ * @author Automattic
+ */
+
+/**
+ * Bail if accessed directly
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Jetpack_WooCommerce_Analytics_Checkout_Flow
+ * Class that handles all page view events for the checkout flow (from product view to order confirmation view)
+ */
+class Jetpack_WooCommerce_Analytics_Checkout_Flow {
+
+	use Jetpack_WooCommerce_Analytics_Trait;
+
+	/**
+	 * Jetpack_WooCommerce_Analytics_Checkout_Flow constructor.
+	 */
+	public function __construct() {
+		$this->find_cart_checkout_content_sources();
+		$this->additional_blocks_on_cart_page     = $this->get_additional_blocks_on_page( 'cart' );
+		$this->additional_blocks_on_checkout_page = $this->get_additional_blocks_on_page( 'checkout' );
+
+		// single product page view.
+		add_action( 'woocommerce_after_single_product', array( $this, 'capture_product_view' ) );
+
+		// order confirmed page view
+		add_action( 'woocommerce_thankyou', array( $this, 'capture_order_confirmation_view' ), 10, 1 );
```

Comment:
Similar to `woocommerce_after_single_product`, I wonder if we should also hook into `wp_footer` here for the same reason. It's very unlikely that the `wp_footer` hook will ever be removed.

Reply:
This has proved to constantly work fine so I didn't want to change it yet, it's also page independent.